### DISCUSSION TITLE
How can I debug tailwindcss?

### DISCUSSION THREAD

**Discussion by caiobukvar** [tailwindlabs/tailwindcss#13015] [Help] [answered]
Hi! I'm having a problem. I'm trying to add next-intl to a project using next and tailwind. The translator works correctly, but after it is set up, tailwind styles are not being applied anymore. 

Below, the inspector to prove it is not applied.
![image](https://github.com/tailwindlabs/tailwindcss/assets/78058375/138af499-0395-4edf-97ff-aa7dd9268378)

Below, how it is supposed to be (branch without next-intl).
![image](https://github.com/tailwindlabs/tailwindcss/assets/78058375/d9aed7a6-9111-4733-b4bc-71a291990836)

tailwind.config.ts

```
import type { Config } from "tailwindcss";

const config: Config = {
  content: [
    "./src/pages/**/*.{js,ts,jsx,tsx,mdx}",
    "./src/components/**/*.{js,ts,jsx,tsx,mdx}",
    "./src/app/**/*.{js,ts,jsx,tsx,mdx}",
  ],
  theme: {...},
  plugins: [],
};
export default config;
```

The project is structured as:
![image](https://github.com/tailwindlabs/tailwindcss/assets/78058375/713206ca-2653-41af-8327-62e7dd652319)

If by some chance anyone knows what is happening, I'd be delighted to hear, lol!

Thanks

**Comment by wongjn** [comment]
Is the Tailwind compilation running on Windows? If so, it could be related to #12718. Otherwise, consider providing a project that reproduces the unexpected behavior.

**Reply by caiobukvar** [reply]
https://github.com/caiobukvar/bkvr-dev/tree/translations

This repo is where the problem is located, branch translations, its public.

The problem is located on [locale]/page.tsx or layout.tsx, don't really know which one renders tailwind styles.

I'm using windows, yes. 
Is it compiling on windows? No idea.

**Reply by caiobukvar** [reply]
If i'm NOT using next-intl, tailwind works just fine, think it's worth saying.

The problem itself appears to happen only when page and layout are inside [locale] folder and not on app/ folder itself.

**Reply by wongjn** [reply]
> I'm using windows, yes.  
> Is it compiling on windows? No idea.

Consider checking the machine you are running the project on to get an idea of what OS it is. There should be a system menu within the machine that should let you know what operating system it is running, i.e Windows, Mac, Linux, etc.

> The problem itself appears to happen only when page and layout are inside [locale] folder and not on app/ folder itself.

Is the Tailwind compilation running on Windows? If so, it could be related to https://github.com/tailwindlabs/tailwindcss/pull/12718.

Also, there doesn't seem to be any files committed to the `src/app/[locale]` folder that I could see on the `translations` branch.

**Reply by caiobukvar** [reply]
Woops, forgot to push stuff into it. Updated now.

**Reply by wongjn** [reply, marked as answer]
You need a PostCSS configuration with `tailwind` added to the `plugins`. You seem to have deleted a `postcss.config.js` file in caiobukvar/bkvr-dev@4683918cc2e7bd9e52d792cc48c71cf55db41485 that would fulfil this requirement – consider restoring this file.

Furthermore, it seems `src/app/_app.tsx` is not in the "main app code", so the `globals.css` is never involved in the site build. Consider `import`ing `src/styles/globals.css` in a different file that *is* in the main app code. For example, Tailwind seems to compile and apply to the page successfully when `import`ed into `src/app/[locale]/layout.tsx`:

```diff
 import Footer from "../components/Footer";
 import Navbar from "../components/Navbar";
+import "../../styles/globals.css";

 export const metadata: Metadata = {
   metadataBase: new URL("https://bkvr.dev.br"),
```
![localhost_3000_en](https://github.com/tailwindlabs/tailwindcss/assets/11310624/30f4bfc6-2f74-47f3-9e57-fa07be7b7874)

**Reply by caiobukvar** [reply]
Perfect!! Thanks for helping, saved me from a bigger headache!
